File: solutions/javascript/isbn-verifier/1/isbn-verifier.js
//
// This is only a SKELETON file for the 'ISBN Verifier' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const isValid = (isbn) => {
  isbn = isbn.replace(/-/g, '');
  if(isbn.length !== 10)
    return false;
  let sum = Number(isbn[9]);
  if(!/\d/.test(isbn[9])) {
    if(isbn[9] === 'X')
      sum = 10;
    else
      return false;
  }
  for(let i = 0; i < isbn.length - 1; i++) {
    if(!/\d/.test(isbn[i]))
      return false;
    sum += Number(isbn[i]) * (10 - i);
  }
  return sum % 11 === 0;
};

File: solutions/python/isbn-verifier/1/isbn_verifier.py
def is_valid(isbn):
    isbn = isbn.replace('-', '')
    len_isbn = len(isbn)
    
    if len_isbn != 10:
        return False
        
    try:
        checksum = 10 if isbn[-1] == "X" else int(isbn[-1])
    except ValueError:
        return False
        
    for index, digit in enumerate(isbn[:-1]):
        try:
            checksum += int(digit) * (len_isbn - index)
        except ValueError:
            return False

    return checksum % 11 == 0